packages/deepseek/src/deepseek-provider.ts

```diff
@@ -70,8 +70,41 @@ export function createDeepSeek(
       `ai-sdk/deepseek/${VERSION}`,
     );
 
+  class DeepSeekChatLanguageModel extends OpenAICompatibleChatLanguageModel {
+    private addJsonInstruction<
+      T extends Parameters<LanguageModelV2['doGenerate']>[0],
+    >(opts: T): T {
+      if (opts.responseFormat?.type !== 'json') return opts;
+
+      const promptArray = Array.isArray(opts.prompt) ? opts.prompt : [];
+
+      const hasJsonWord = promptArray.some(
+        m =>
+          m.role === 'user' &&
+          Array.isArray(m.content) &&
+          m.content.some(p => p.type === 'text' && /json/i.test(p.text ?? '')),
+      );
+      if (hasJsonWord) return opts;
```

Comment:
is this check needed? Or can we simply always add the additional instruction below